From c5cccc0a82dfa278d75019646fae64771dcedcb3 Mon Sep 17 00:00:00 2001
From: Iain Lane <iainl@gnome.org>
Date: Fri, 23 Nov 2018 10:23:47 +0000
Subject: [PATCH 40/55] Show the 'link' emblem when a desktop file is
 executable

This is to give some visual indication that the icon is some sort of
link.
---
 fileItem.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/fileItem.js b/fileItem.js
index a988c51..5deb846 100644
--- a/fileItem.js
+++ b/fileItem.js
@@ -377,11 +377,14 @@ var FileItem = class {
             }
         }
         let itemIcon = Gio.EmblemedIcon.new(icon, null);
+
         if (this._isSymlink) {
             if (this._isBrokenSymlink)
                 itemIcon.add_emblem(Gio.Emblem.new(Gio.ThemedIcon.new('emblem-unreadable')));
             else
                 itemIcon.add_emblem(Gio.Emblem.new(Gio.ThemedIcon.new('emblem-symbolic-link')));
+        } else if (this.trustedDesktopFile) {
+            itemIcon.add_emblem(Gio.Emblem.new(Gio.ThemedIcon.new('emblem-symbolic-link')));
         }
 
         return new St.Icon({ gicon: itemIcon,
@@ -646,7 +649,7 @@ var FileItem = class {
     }
 
     get trustedDesktopFile() {
-        return this._isDesktopFile && this._attributeCanExecute && this._trusted;
+        return this._isDesktopFile && this._attributeCanExecute && this.metadataTrusted;
     }
 
     get displayName() {
-- 
2.20.0

